### DISCUSSION TITLE
Using HTML in JointJS+ elements?

### DISCUSSION THREAD

**Discussion by taeil00** [clientIO/joint#2054] [Q&A]
Hello,
I'm trying to HTML elements inside my JointJS+ graphs. But there's no shapes.html element.
Can't I use the shapes.html element anymore?
I saw this page  https://resources.jointjs.com/tutorial/html-elements
and the article said "Disclaimer - The following tutorial was created with a past version of JointJS. "
Is shapes.html element excluded from the JointJS+ ?
Is there any alternative?

*I need to draw rendered html of MS Appcard on jointJs+ graph.

**Comment by kumilingus** [comment]
The `shapes.html` elements were never part of JointJS. It was just an example/tutorial.

A newer example of HTML shapes is to be found here:
https://www.jointjs.com/demos/tasks

An alternative approach is to use `<foreignObject>`:
https://github.com/clientIO/joint/blob/master/demo/shapes/src/foreign-object.js

**Reply by taeil00** [reply]
Thank you for your reply.
Source code of the url 'https://www.jointjs.com/demos/tasks' also used joint.shapes.html.ElementView.
ex>
    joint.shapes.html.ElementView = ElementView.extend({ ...

But there's no joint.shapes.**html** element now in JointJS+.

**Reply by kumilingus** [reply]
For the demo purposes only, the `joint.shapes.html.Element` model is defined [here](https://github.com/clientIO/joint/blob/master/demo/html/src/joint.shapes.html.js#L6-L107).

> But there's no joint.shapes.html element now in JointJS+.

There is not and never has been `joint.shapes.html element` in JointJS+.

**Reply by taeil00** [reply]
Oh I see, It was just a custom element !
Thank you for your kind answer.
